enhancements/network/multus-services.md

```diff
@@ -0,0 +1,239 @@
+
+- `label: service.kubernetes.io/service-proxy-name` (well-known label, defined in kubernetes)
+
+Users will need to set the label `service.kubernetes.io/service-proxy-name`, with the value, `multus-proxy`. This
+label specifies the component that takes care of the service. Without this variable, upstream kubernetes
+(i.e. kube-proxy) will handle this service, then kube-proxy will create iptables rules. This label should be set to
+`multus-proxy`. For OpenShift, we should verify that this label is recognized by openshift-sdn as well as
+ovn-kubernetes. This enhancement should address openshift-sdn and ovn-kubernetes repositories.
+
+- annotations: k8s.v1.cni.cncf.io/service-network (newly introduced)
+
+`k8s.v1.cni.cncf.io/service-network` specifies which network (i.e. net-attach-def) is the target for exposing the
+service. This label specifies only one net-attach-def in same namespace of the Service/Pods, hence Service/Network
+Attachment Definition/Pods should be in the same namespace. This annotation only allows users to specify one network,
+hence one Service VIP should match one of service/network mappings.
+
+Then multus service controller will watch the service and pods.
+
+2. Pods are launched and multus service controller creates the endpointslices for the sevice.
+
+Once multus service controller finds pods associated with the service, then the controller will create EndpointSlices
+for the service. The EndpointSlices contains a pod's net-attach-def IP address from pod's network status annotation
+and the EndpointSlices has same label, `service.kubernetes.io/service-proxy-name`, and its value from corresponding
+service to avoid kube-proxy to add iptables rules for Kubernetes Service forwarding plane. When the pods/services
+is updated/removed, then multus service controller changes/removes the corresponding endpoitslices to track the
+changes.
+
+3. Multus proxy generates iptables rules for the service in Pod's network namespace.
+
+Multus proxy watches the endpointslices and services. Periodically multus proxy generates iptables rules for the pods
+and put it in a pod's network namespace (not host network namespace, as kube-proxy does). Multus proxy tracks the
+changes of service/endpoints, hence multus proxy update/remove iptables rules when service/endpoints are
+updated/removed.
+
+### Risks and Mitigations
+
+#### Interworking service.kubernetes.io/service-proxy-name among other Kubernetes network
+
+This feature depends on the well-known label, `service.kubernetes.io/service-proxy-name` implementation. If some
+Kubernetes network solution satisfies the following condition, then this feature does not work:
+
+- The SDN solution provides forwarding plane for Kubernetes service (i.e. Cilium, ovn-kubernetes).
+- The SDN solution does not recognize the well-known label, `service.kubernetes.io/service-proxy-name`.
+
+Hence we should mention which SDN solution supports this enhancement in documents.
```

Comment:
and hopefully make Issues to networking plugins that do not respect the label. :)

Reply:
agree, but it could be the future work (i.e. not for the 4.10 release).